services/convert/user.go

```diff
@@ -41,12 +35,12 @@ func ToUserWithAccessMode(ctx context.Context, user *user_model.User, accessMode
 	if user == nil {
 		return nil
 	}
-	return toUser(ctx, user, accessMode != perm.AccessModeNone, false)
+	return toUser(ctx, user, nil, accessMode)
 }
 
 // toUser convert user_model.User to api.User
-// signed shall only be set if requester is logged in. authed shall only be set if user is site admin or user himself
-func toUser(ctx context.Context, user *user_model.User, signed, authed bool) *api.User {
+// accessMode is only used if doer is nil
+func toUser(ctx context.Context, user, doer *user_model.User, accessMode perm.AccessMode) *api.User {
```

Comment:
Looks like `accessMode` is always `perm.AccessModeNone`, maybe we should remove it.

Reply:
If `doer` is nil, we use `accessMode` to check whether provide `Email` info.